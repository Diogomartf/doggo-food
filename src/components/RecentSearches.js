import React, { useState } from 'react';
import styled from 'styled-components';
import { X, Menu } from 'react-feather';

const RecentSearches = ({ searches, deleteRecentSearch, onClick }) => {
  const [expand, setExpand] = useState(false);
  return (
    searches.length > 0 && (
      <SearchWrapper>
        <Container>
          <CollapseWrapper onClick={() => setExpand(!expand)}>
            <Menu size={13} />
            <CollapseTitle>Recent searches</CollapseTitle>
          </CollapseWrapper>
        </Container>
        {expand && (
          <Box>
            <Container>
              {searches.map((search, index) => (
                <SearchItem
                  key={index}
                  onClick={() => {
                    onClick(search);
                  }}
                >
                  <div>{search}</div>
                  <CloseIcon
                    sizer={12}
                    onClick={e => {
                      e.stopPropagation();
                      deleteRecentSearch(index);
                    }}
                  />
                </SearchItem>
              ))}
            </Container>
          </Box>
        )}
      </SearchWrapper>
    )
  );
};

const SearchWrapper = styled.div`
  width: 100%;
  background-color: ${props => props.theme.colors.lightGray};
`;

const Container = styled.div`
  padding: 0 24px 12px 24px;
`;

const CollapseWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-end;
  width: 100%;
  cursor: pointer;
`;

const CollapseTitle = styled.div`
  margin-left: 4px;
  color: ${props => props.theme.colors.darkGray};
  text-transform: uppercase;
  font-size: 12px;
`;

const Box = styled.div`
  position: absolute;
  width: 100%;
  z-index: 1;
  background-color: ${props => props.theme.colors.lightGray};
`;

const CloseIcon = styled(X)`
  color: ${props => props.theme.colors.darkGray};
`;

const SearchItem = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;

  height: 38px;
  padding: 0 12px;

  background-color: ${props => props.theme.colors.lightGray};
  cursor: pointer;

  :hover {
    background-color: ${props => props.theme.colors.gray};
    border-radius: 4px;
  }
`;

export default RecentSearches;
