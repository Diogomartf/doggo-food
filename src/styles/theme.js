import colors from './colors';

export default {
  name: 'default theme',
  colors
};
