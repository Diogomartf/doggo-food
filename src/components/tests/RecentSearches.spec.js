import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../../styles';

import RecentSearches from '../RecentSearches';

describe('<RecentSearches/>', () => {
  const deleteRecentSearchMock = jest.fn();
  const onClickRecentSearchMock = jest.fn();

  it('should render correctly', () => {
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <RecentSearches
          searches={['pesto', 'cheese']}
          deleteRecentSearch={deleteRecentSearchMock}
          onClick={onClickRecentSearchMock}
        />
      </ThemeProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render if has no searches', () => {
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <RecentSearches
          searches={[]}
          deleteRecentSearch={deleteRecentSearchMock}
          onClick={onClickRecentSearchMock}
        />
      </ThemeProvider>
    );
    expect(wrapper.find('SearchWrapper')).toHaveLength(0);
  });

  it('should render box when expanded', () => {
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <RecentSearches
          searches={['pesto', 'pizza']}
          deleteRecentSearch={deleteRecentSearchMock}
          onClick={onClickRecentSearchMock}
        />
      </ThemeProvider>
    );
    wrapper.find('CollapseWrapper').simulate('click');
    expect(wrapper.find('Box')).toHaveLength(1);
  });
});
