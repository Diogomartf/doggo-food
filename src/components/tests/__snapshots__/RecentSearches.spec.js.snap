// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecentSearches/> should render correctly 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "black": "#2B2B2B",
        "darkGray": "#626262",
        "gray": "#EAEAEA",
        "lightGray": "#F4F4F4",
        "primary": "#6056F5",
        "red": "#CF0A2C",
        "white": "#FFFFFF",
        "yellow": "#FBB03B",
      },
      "name": "default theme",
    }
  }
>
  <RecentSearches
    deleteRecentSearch={[MockFunction]}
    onClick={[MockFunction]}
    searches={
      Array [
        "pesto",
        "cheese",
      ]
    }
  >
    <SearchWrapper>
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": false,
              "lastClassName": "mpKfW",
              "rules": Array [
                "
  width: 100%;
  background-color: ",
                [Function],
                ";
",
              ],
            },
            "displayName": "SearchWrapper",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <div
          className="sc-bdVaJa mpKfW"
        >
          <styled.div>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": true,
                    "lastClassName": "hYVoYR",
                    "rules": Array [
                      "
  padding: 0 24px 12px 24px;
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="sc-bwzfXH hYVoYR"
              >
                <CollapseWrapper
                  onClick={[Function]}
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-htpNat",
                          "isStatic": true,
                          "lastClassName": "emitFB",
                          "rules": Array [
                            "
  display: flex;
  align-items: center;
  justify-content: flex-end;
  width: 100%;
  cursor: pointer;
",
                          ],
                        },
                        "displayName": "CollapseWrapper",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-htpNat",
                        "target": "div",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    onClick={[Function]}
                  >
                    <div
                      className="sc-htpNat emitFB"
                      onClick={[Function]}
                    >
                      <Menu
                        color="currentColor"
                        size={13}
                      >
                        <svg
                          fill="none"
                          height={13}
                          stroke="currentColor"
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeWidth="2"
                          viewBox="0 0 24 24"
                          width={13}
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <line
                            x1="3"
                            x2="21"
                            y1="12"
                            y2="12"
                          />
                          <line
                            x1="3"
                            x2="21"
                            y1="6"
                            y2="6"
                          />
                          <line
                            x1="3"
                            x2="21"
                            y1="18"
                            y2="18"
                          />
                        </svg>
                      </Menu>
                      <styled.div>
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-bxivhb",
                                "isStatic": false,
                                "lastClassName": "iLjUPw",
                                "rules": Array [
                                  "
  margin-left: 4px;
  color: ",
                                  [Function],
                                  ";
  text-transform: uppercase;
  font-size: 12px;
",
                                ],
                              },
                              "displayName": "styled.div",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-bxivhb",
                              "target": "div",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                        >
                          <div
                            className="sc-bxivhb iLjUPw"
                          >
                            Recent searches
                          </div>
                        </StyledComponent>
                      </styled.div>
                    </div>
                  </StyledComponent>
                </CollapseWrapper>
              </div>
            </StyledComponent>
          </styled.div>
        </div>
      </StyledComponent>
    </SearchWrapper>
  </RecentSearches>
</ThemeProvider>
`;
